jax/interpreters/pxla.py

```diff
@@ -1172,8 +1117,13 @@ def partitioned_sharding_spec(num_partitions: int,
         mesh_mapping=maybe_replicate)
   else:
     assert len(partitions) == len(aval.shape)
-    return ShardingSpec(sharding=map(Chunked, partitions),
-                        mesh_mapping=map(ShardedAxis, range(len(partitions))))
+    # Chunked expects a list of integers and never an integer.
+    partitions_for_chunked = [
+        [x] if isinstance(x, int) else x for x in partitions
+    ]
```

Comment:
nit: the whitespace in here is a bit excessive. We don't forcefully follow Google conventions for line length, and IMO it is more readable if you fold it. But not a hard constraint

Reply:
Also, how would `x` be anything else than an `int`? The type annotation explicitly rules it out.

Reply:
Done. (it's just that I am not really sure the type annotation is correctly enforced here.).